[Filebeat][O365 Module] client_secret produces error if secret starts with @

Seems like there might be an issue with the configuration mapping (mustache?) for certain fields starting with `@`.

Looking at: https://discuss.elastic.co/t/yaml-found-character-that-cannot-start-any-token-when-value-starts-with/269778

The following example:
```
- module: o365
  audit:
    enabled: true
    var.application_id: "xxx"
    var.tenants:
      - id: "xxx"
        name: "xxx"
    var.client_secret: "@/xxx"
```

Would produce an error like:
`cfgfile/list.go:99 Error creating runner from config: Error getting config for fileset o365/audit: Error reading input config: yaml: line 12: found character that cannot start any token.`

Will update this issue once I have looked into it.

--- a/CHANGELOG.next.asciidoc
+++ b/CHANGELOG.next.asciidoc
@@ -287,6 +287,7 @@ https://github.com/elastic/beats/compare/v7.0.0-alpha2...master[Check the HEAD d
 - Fix integer overflow in S3 offsets when collecting very large files. {pull}22523[22523]
 - Fix CredentialsJSON unpacking for `gcp-pubsub` and `httpjson` inputs. {pull}23277[23277]
 - Strip Azure Eventhub connection string in debug logs. {pulll}25066[25066]
+- Fix o365 module config when client_secret contains special characters. {issue}25058[25058]
 
 *Filebeat*
 

--- a/filebeat/fileset/fileset.go
+++ b/filebeat/fileset/fileset.go
@@ -300,8 +300,11 @@ func getTemplateFunctions(vars map[string]interface{}) (template.FuncMap, error)
 			return false
 		},
 		"tojson": func(v interface{}) (string, error) {
-			bytes, err := json.Marshal(v)
-			return string(bytes), err
+			var buf strings.Builder
+			enc := json.NewEncoder(&buf)
+			enc.SetEscapeHTML(false)
+			err := enc.Encode(v)
+			return buf.String(), err
 		},
 		"IngestPipeline": func(shortID string) string {
 			return formatPipelineID(

--- a/x-pack/filebeat/module/o365/audit/config/input.yml
+++ b/x-pack/filebeat/module/o365/audit/config/input.yml
@@ -1,15 +1,15 @@
 {{ if eq .input "o365audit" }}
 
 type: o365audit
-{{ if .application_id }}application_id: {{ .application_id }}{{ end }}
+{{ if .application_id }}application_id: '{{ .application_id }}'{{ end }}
 tenant_id:
 {{ range .tenants }}
- - {{ .id }}
+ - '{{ .id }}'
 {{ end }}
-{{ if .certificate }}certificate: {{ .certificate }}{{ end }}
-{{ if .key }}key: {{ .key }}{{ end }}
-{{ if .key_passphrase }}key_passphrase: {{ .key_passphrase }}{{ end }}
-{{ if .client_secret }}client_secret: {{ .client_secret }}{{ end }}
+{{ if .certificate }}certificate: '{{ .certificate }}'{{ end }}
+{{ if .key }}key: '{{ .key }}'{{ end }}
+{{ if .key_passphrase }}key_passphrase: '{{ .key_passphrase }}'{{ end }}
+{{ if .client_secret }}client_secret: '{{ .client_secret }}'{{ end }}
 {{ if eq "string" (printf "%T" .content_type) }}
 content_type: {{ .content_type }}
 {{ else }}

